core/router/uplink.go

```diff
@@ -102,11 +110,16 @@ func (r *router) HandleUplink(gatewayID string, uplink *pb.UplinkMessage) (err e
 
 	if len(brokers) == 0 {
 		ctx.Debug("No brokers to forward message to")
+		uplink.Trace = uplink.Trace.WithEvent(trace.DropEvent, "reason", "no brokers")
```

Comment:
Be consistent with keys; e.g. don't mix `reason` and `error` for drop event. I'd suggest using `reason` in case of the drop event also if it's an error. It will be easier for us to rely on metadata keys for specific event types.